controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java

```diff
@@ -718,6 +718,7 @@ public void deleteApplication(TenantAndApplicationId id, Optional<Credentials> c
         if (instances.size() > 1)
             throw new IllegalArgumentException("Could not delete application; more than one instance present: " + instances);
 
+        lockApplicationOrThrow(id, application -> store(application.with(DeploymentSpec.empty)));
```

Comment:
Hopefully not a long-lived solution. 🙂

Reply:
Hehe.

Reply:
No, this all needs to be solved properly. Deletion at application level, with a confirmatino dialogue with a list of all instances to be deleted, or something.